<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>potentiometerEditForm</class>
 <widget class="QWidget" name="potentiometerEditForm">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>832</width>
    <height>674</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <property name="sizeConstraint">
    <enum>QLayout::SetDefaultConstraint</enum>
   </property>
   <item row="8" column="0" colspan="6">
    <layout class="QGridLayout" name="gridLayout_3">
     <item row="2" column="1">
      <layout class="QHBoxLayout" name="horizontalLayout_4">
       <property name="spacing">
        <number>2</number>
       </property>
       <item>
        <widget class="QLabel" name="label_11">
         <property name="text">
          <string>Set Dataref: </string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="ADDDREF_BTN">
         <property name="text">
          <string>...</string>
         </property>
         <property name="icon">
          <iconset resource="resources.qrc">
           <normaloff>:/newPrefix/plusIcon.png</normaloff>:/newPrefix/plusIcon.png</iconset>
         </property>
         <property name="autoRaise">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="RMDREF_BTN">
         <property name="text">
          <string>...</string>
         </property>
         <property name="icon">
          <iconset resource="resources.qrc">
           <normaloff>:/newPrefix/minusIcon.png</normaloff>:/newPrefix/minusIcon.png</iconset>
         </property>
         <property name="autoRaise">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_2">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
     <item row="2" column="4">
      <layout class="QHBoxLayout" name="horizontalLayout_5"/>
     </item>
     <item row="3" column="1">
      <widget class="QTableWidget" name="DREFS_TABLE">
       <column>
        <property name="text">
         <string>Dataref</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Index</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Set to Value</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Type</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Unit</string>
        </property>
       </column>
      </widget>
     </item>
     <item row="3" column="3" colspan="2">
      <widget class="QLabel" name="drefsHelpText">
       <property name="minimumSize">
        <size>
         <width>250</width>
         <height>0</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>300</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="text">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Enter series of points [pot input value, output dataref value]. The output value of the dataref in between 2 points will be linearly interpolated. &lt;/p&gt;&lt;p&gt;Note that the potentiometer value varies between 0 and 1023 (this is the raw output reading from the ADC of the arduino)&lt;/p&gt;&lt;p&gt;For example to set the dataref to vary linearly between 10 and 300 when the potentiometer value varies between 0 and 1023, enter: &lt;/p&gt;&lt;p&gt;&lt;span style=&quot; color:#0000ff;&quot;&gt;[0,10], [1023,300]&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="textFormat">
        <enum>Qt::RichText</enum>
       </property>
       <property name="alignment">
        <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
       </property>
       <property name="wordWrap">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="3" column="2" colspan="3">
    <widget class="QLineEdit" name="nameLineEdit">
     <property name="minimumSize">
      <size>
       <width>250</width>
       <height>0</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>350</width>
       <height>16777215</height>
      </size>
     </property>
    </widget>
   </item>
   <item row="10" column="0" colspan="6">
    <layout class="QGridLayout" name="gridLayout_4">
     <item row="6" column="0">
      <widget class="QTableWidget" name="CMDS_TABLE">
       <property name="toolTip">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Enter intervals where the command should be sent. &lt;/p&gt;&lt;p&gt;For example to send the command if the pot value is between 0 and 200 or between 600 and 800, enter: &lt;/p&gt;&lt;p&gt;&lt;span style=&quot; color:#0000ff;&quot;&gt;(0,200), (600,800)&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="toolTipDuration">
        <number>3</number>
       </property>
       <column>
        <property name="text">
         <string>Command</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Send if pot value is in intervals</string>
        </property>
        <property name="toolTip">
         <string extracomment="Enter intervals where the command should be sent. For example to send the command if the pot value is between 0 and 200 or between 600 and 800, enter (0,200), (600,800)">&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Enter intervals where the command should be sent. &lt;/p&gt;&lt;p&gt;For example to send the command if the pot value is between 0 and 200 or between 600 and 800, enter: &lt;/p&gt;&lt;p&gt;&lt;span style=&quot; color:#0000ff;&quot;&gt;(0,200), (600,800)&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </column>
      </widget>
     </item>
     <item row="3" column="0">
      <layout class="QHBoxLayout" name="horizontalLayout_6">
       <item>
        <widget class="QLabel" name="label_12">
         <property name="text">
          <string>Send XPlane Command: </string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="ADDCMD_BTN">
         <property name="text">
          <string>...</string>
         </property>
         <property name="icon">
          <iconset>
           <normalon>:/newPrefix/plusIcon.png</normalon>
          </iconset>
         </property>
         <property name="autoRaise">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="RMCMD_BTN">
         <property name="text">
          <string>...</string>
         </property>
         <property name="icon">
          <iconset>
           <normalon>:/newPrefix/minusIcon.png</normalon>
          </iconset>
         </property>
         <property name="autoRaise">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_3">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
     <item row="3" column="1">
      <layout class="QHBoxLayout" name="horizontalLayout_7"/>
     </item>
     <item row="6" column="1">
      <widget class="QLabel" name="label">
       <property name="minimumSize">
        <size>
         <width>250</width>
         <height>0</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>300</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="text">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Enter intervals where the command should be sent. &lt;/p&gt;&lt;p&gt;For example to send the command if the pot value is between 0 and 200 or between 600 and 800, enter: &lt;/p&gt;&lt;p&gt;&lt;span style=&quot; color:#0000ff;&quot;&gt;[0,200], [600,800]&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="textFormat">
        <enum>Qt::RichText</enum>
       </property>
       <property name="alignment">
        <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
       </property>
       <property name="wordWrap">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="4" column="2" colspan="3">
    <widget class="QLineEdit" name="IDlineEdit">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="minimumSize">
      <size>
       <width>250</width>
       <height>0</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>350</width>
       <height>16777215</height>
      </size>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="5" column="2">
    <widget class="QComboBox" name="PIN_comboBox">
     <property name="maximumSize">
      <size>
       <width>50</width>
       <height>16777215</height>
      </size>
     </property>
    </widget>
   </item>
   <item row="0" column="0" colspan="6">
    <widget class="QLabel" name="label_8">
     <property name="font">
      <font>
       <pointsize>10</pointsize>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="styleSheet">
      <string notr="true">color: white;
background: rgb(0, 151, 157);
padding: 3px
</string>
     </property>
     <property name="text">
      <string>Edit potentiometer</string>
     </property>
    </widget>
   </item>
   <item row="6" column="0">
    <spacer name="verticalSpacer_2">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeType">
      <enum>QSizePolicy::Fixed</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>10</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="1" column="0">
    <widget class="QLabel" name="label_17">
     <property name="text">
      <string>Value</string>
     </property>
    </widget>
   </item>
   <item row="9" column="0">
    <widget class="QLabel" name="label_15">
     <property name="font">
      <font>
       <pointsize>9</pointsize>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>Command actions:</string>
     </property>
    </widget>
   </item>
   <item row="3" column="0">
    <widget class="QLabel" name="label_3">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="layoutDirection">
      <enum>Qt::LeftToRight</enum>
     </property>
     <property name="text">
      <string>Name</string>
     </property>
    </widget>
   </item>
   <item row="7" column="0">
    <widget class="QLabel" name="label_9">
     <property name="font">
      <font>
       <pointsize>9</pointsize>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>Dataref Actions:</string>
     </property>
    </widget>
   </item>
   <item row="5" column="0">
    <widget class="QLabel" name="label_18">
     <property name="text">
      <string>Arduino pin:</string>
     </property>
    </widget>
   </item>
   <item row="4" column="0">
    <widget class="QLabel" name="label_19">
     <property name="text">
      <string>ID</string>
     </property>
    </widget>
   </item>
   <item row="12" column="0">
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>26</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="1" column="4">
    <widget class="QSlider" name="valueSlider">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="maximumSize">
      <size>
       <width>200</width>
       <height>16777215</height>
      </size>
     </property>
     <property name="styleSheet">
      <string notr="true">QSlider::groove:horizontal {
    border: 1px solid #999999;
    height: 8px; /* the groove expands to the size of the slider by default. by giving it a height, it has a fixed size */
    background: qlineargradient(x1:0, y1:0, x2:0, y2:1, stop:0 #B1B1B1, stop:1 #c4c4c4);
    margin: 2px 0;
}

QSlider::handle:horizontal {
    background: qlineargradient(x1:0, y1:0, x2:1, y2:1, stop:0 #b4b4b4, stop:1 #8f8f8f);
    border: 1px solid #5c5c5c;
    width: 5px;
    margin: -2px 0; /* handle is placed by default on the contents rect of the groove. Expand outside the groove */
    border-radius: 2px;
}

QSlider::sub-page:horizontal{
	border: 1px solid #999999;
    height: 8px;     
	background: qlineargradient(x1:0, y1:0, x2:0, y2:1, stop:0 #00979d, stop:1 #007579);
    margin: 2px 0;
}</string>
     </property>
     <property name="maximum">
      <number>1023</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="invertedAppearance">
      <bool>false</bool>
     </property>
     <property name="tickPosition">
      <enum>QSlider::TicksAbove</enum>
     </property>
     <property name="tickInterval">
      <number>128</number>
     </property>
    </widget>
   </item>
   <item row="1" column="2">
    <widget class="QLabel" name="valueLabel">
     <property name="minimumSize">
      <size>
       <width>40</width>
       <height>0</height>
      </size>
     </property>
     <property name="text">
      <string>0</string>
     </property>
    </widget>
   </item>
   <item row="1" column="5">
    <spacer name="horizontalSpacer_5">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>40</width>
       <height>20</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>ADDDREF_BTN</sender>
   <signal>clicked()</signal>
   <receiver>potentiometerEditForm</receiver>
   <slot>addDataref()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>98</x>
     <y>207</y>
    </hint>
    <hint type="destinationlabel">
     <x>381</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>RMDREF_BTN</sender>
   <signal>clicked()</signal>
   <receiver>potentiometerEditForm</receiver>
   <slot>rmDataref()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>123</x>
     <y>207</y>
    </hint>
    <hint type="destinationlabel">
     <x>456</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ADDCMD_BTN</sender>
   <signal>clicked()</signal>
   <receiver>potentiometerEditForm</receiver>
   <slot>addCommand()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>141</x>
     <y>433</y>
    </hint>
    <hint type="destinationlabel">
     <x>119</x>
     <y>4</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>RMCMD_BTN</sender>
   <signal>clicked()</signal>
   <receiver>potentiometerEditForm</receiver>
   <slot>rmCommand()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>174</x>
     <y>437</y>
    </hint>
    <hint type="destinationlabel">
     <x>166</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>CMDS_TABLE</sender>
   <signal>cellDoubleClicked(int,int)</signal>
   <receiver>potentiometerEditForm</receiver>
   <slot>editXPCommand()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>240</x>
     <y>497</y>
    </hint>
    <hint type="destinationlabel">
     <x>238</x>
     <y>5</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>nameLineEdit</sender>
   <signal>editingFinished()</signal>
   <receiver>potentiometerEditForm</receiver>
   <slot>updateXMLdata()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>368</x>
     <y>74</y>
    </hint>
    <hint type="destinationlabel">
     <x>510</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>PIN_comboBox</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>potentiometerEditForm</receiver>
   <slot>updateXMLdata()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>183</x>
     <y>141</y>
    </hint>
    <hint type="destinationlabel">
     <x>487</x>
     <y>-6</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>CMDS_TABLE</sender>
   <signal>itemChanged(QTableWidgetItem*)</signal>
   <receiver>potentiometerEditForm</receiver>
   <slot>updateXMLdata()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>360</x>
     <y>481</y>
    </hint>
    <hint type="destinationlabel">
     <x>563</x>
     <y>44</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>PIN_comboBox</sender>
   <signal>currentTextChanged(QString)</signal>
   <receiver>potentiometerEditForm</receiver>
   <slot>updatePin()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>183</x>
     <y>141</y>
    </hint>
    <hint type="destinationlabel">
     <x>344</x>
     <y>163</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>DREFS_TABLE</sender>
   <signal>cellDoubleClicked(int,int)</signal>
   <receiver>potentiometerEditForm</receiver>
   <slot>editXPDataref()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>666</x>
     <y>346</y>
    </hint>
    <hint type="destinationlabel">
     <x>665</x>
     <y>99</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>DREFS_TABLE</sender>
   <signal>itemChanged(QTableWidgetItem*)</signal>
   <receiver>potentiometerEditForm</receiver>
   <slot>updateXMLdata()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>595</x>
     <y>317</y>
    </hint>
    <hint type="destinationlabel">
     <x>571</x>
     <y>150</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>addCommand()</slot>
  <slot>rmCommand()</slot>
  <slot>addDataref()</slot>
  <slot>rmDataref()</slot>
  <slot>editXPCommand()</slot>
  <slot>updateXMLdata()</slot>
  <slot>testCommands()</slot>
  <slot>updatePin()</slot>
  <slot>editXPDataref()</slot>
  <slot>testDatarefs()</slot>
  <slot>activateSave()</slot>
 </slots>
</ui>
